collections.js

```diff
@@ -118,8 +118,49 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Queue, a simple queue implementation.
+// The head of the queue is at the beginning of the backing array.
+
+function Queue() {
+  this.head_ = 0;
+  this.headLimit_ = 1000;
+  this.sliceMin_ = 10;
+  this.q_ = [];
 }
 
+Queue.prototype = {
+  enqueue: function(item) {
+    this.q_.push(item);
+    return this;
+  },
+
+  dequeue: function() {
+    var result = this.q_[this.head_];
+    delete this.q_[this.head_];
```

Comment:
Just set this.q_[this.head_] to null or undefined. We don't need to reconfigure the array. I think this would have performance implications.